pmd-java/src/main/java/net/sourceforge/pmd/lang/java/rule/bestpractices/CheckResultSetRule.java

```diff
@@ -56,12 +58,14 @@ private boolean isResultSetMethod(ASTMethodCall node) {
     }
 
     private boolean isCheckedIndirectly(ASTMethodCall node) {
-        if (!(node.getParent() instanceof ASTVariableDeclarator)) {
+        final NodeStream<ASTVariableDeclarator> variableDeclarators = node.ancestors(ASTVariableDeclarator.class);
+        if (variableDeclarators.isEmpty()) {
```

Comment:
I'm not really sure whether this is the right way to find more indirect usages. While this fixes the FP, it also introduces new FNs (see 2 removed violations from regression tester).

Is there any additional filtering or condition that I can add to improve the search for indirect usages?